enhancements/security/selinux-operator.md

```diff
@@ -0,0 +1,269 @@
+#### Module removal
+
+As a system administrator I want to remove the custom SELinux policy once I
+no longer run the workload in the cluster.
+
+#### Auditing
+
+As a system administrator I want to know what custom policies are installed all
+over the cluster.
+
+#### Security (RBAC)
+
+As a system administrator, I want to enforce that only certain users can create
+policies in certain namespaces.
+
+This way, only trusted users would be able to install policies in the system.
+
+#### Security (namespace separation)
+
+As a system administrator, I want to make sure that workloads can only use
+approved policies.
+
+This way, an attacker can't take into use a policy from a namespace that's not
+accessible.
+
+### API Specification
+
+This operator would expose a CRD that would look as follows:
+
+```
+apiVersion: selinux.openshift.io/v1alpha1
+kind: SelinuxPolicy
+metadata:
+  name: virt-launcher
+  namespace: my-namespace
+spec:
+  apply: true
+  policy: |
+    (blockinherit container)
+    (typeattributeset sandbox_net_domain (process))
+    (allow process kernel_t (system (module_request)))
+    (allow process mtrr_device_t (file (write)))
+    (allow process self (tun_socket (relabelfrom)))
+    (allow process self (tun_socket (relabelto)))
+    (allow process self (tun_socket (attach_queue)))
+    (allow process sysfs_t (file (write)))
+    (allow process tmp_t (dir (write add_name open getattr setattr read link search remove_name reparent lock ioctl)))
+    (allow process tmp_t (file (setattr open read write create getattr append ioctl lock)))
+    (allow process container_share_t (dir (write add_name)))
+    (allow process container_share_t (file (create setattr write )))
+    (allow process container_var_run_t (dir (write add_name open getattr setattr read link search remove_name reparent lock ioctl)))
+    (allow process container_var_run_t (file (setattr open read write create getattr append ioctl lock)))
+status:
+  usage: virt-launcher_my-namespace.process
+  state: INSTALLED
+```
+
+Where:
+
+* `spec.apply`: Is a flag that allows us to tell the operator if the policy is to be
+  applied (installed) or not. Having this flag allows policy writers and system
+  administrators to verify the policy before taking it into use in the system.
+  Automated tools such as the aforementioned `selinux-policy-helper-operator`
+  would generate policies with this flag set to `false`.
+* `spec.policy`: Contains the actual lines of policy in
+  [CIL language](https://github.com/SELinuxProject/cil/wiki). The decision for
+  this language is because `Udica` already writes policies in such a language,
+  and it's an easier to read format than the typical SELinux type enforcement
+  (te) language.
+* `status.usage`: This string indicates how the user is supposed to use this
+  policy in their `Pod` templates. The format is as follows:
+  `<policy name>_<policy namespace>.process>`
+* `status.state`: This represents the current status of the policy in the
```

Comment:
what does status.state mean if the cluster is autoscaled?  is the state actually installed until the the daemonset is scheduled to that new node and rendering its value?  if this was driven through MCO, we have a way of knowing if a change is applied across a whole pool of nodes.

Reply:
The installation is managed via a DaemonSet. If you scale your nodes, the daemonset scales too. so those nodes should get the policy as well.